chunk_arrow.go

```diff
@@ -34,17 +40,22 @@ func (arc *arrowResultChunk) decodeArrowChunk(rowType []execResponseRowType, hig
 		columns := record.Columns()
 		tmpRows := make([]chunkRowType, numRows)
 
-		for colIdx, col := range columns {
-			destcol := make([]snowflakeValue, numRows)
-			if err = arrowToValue(&destcol, rowType[colIdx], col, highPrec); err != nil {
-				return nil, err
-			}
+		if isArrowRecordChanEnabled(arc.ctx) {
```

Comment:
as mentioned in person, let's try to make another method `decodeArrowRecords` that handles this use case